### ISSUE TITLE
GNOME module doesn't use PKG_CONFIG_PATH to find dependencies

### ISSUE THREAD

**Issue by lantw44** [mesonbuild/meson#5643]
In mesonbuild/modules/gnome.py, there are two places using `PkgConfigDependency`:
```python
    def _get_native_glib_version(state):
        global native_glib_version 
        if native_glib_version is None:
            glib_dep = PkgConfigDependency('glib-2.0', state.environment,
                                           {'native': True, 'required': False})
```
```python
    def _get_gir_dep(self, state):
        try:
            gir_dep = self.gir_dep or PkgConfigDependency('gobject-introspection-1.0',
                                                          state.environment,
                                                          {'native': True})
```

I don't know why they both use `'native': True`. The use of `native` causes `for_machine` of `PkgConfigDependency` to be set to `BUILD` instead of the default value `HOST`. Since meson only initializes `pkg_config_path` option from `PKG_CONFIG_PATH` environment variable for `HOST`, the use of `BUILD` here causes meson to run pkg-config with an empty `PKG_CONFIG_PATH`. Therefore, if your `gobject-introspection-1.0` requires `PKG_CONFIG_PATH` to find, it fails unless you if you remember to pass `--build.pkg-config-path` to meson.

This problem causes folks to fail to build on my computer. I have `PKG_CONFIG_PATH` properly set by JHBuild for development, but meson ignores it and complains that my `gobject-introspection-1.0` is too old.

The version installed in JHBuild environment should be new enough.
```
$ pkg-config --modversion gobject-introspection-1.0
1.61.1
$ PKG_CONFIG_PATH= pkg-config --modversion gobject-introspection-1.0
1.56.1
```

But meson just ignored it.
```
Found pkg-config: /usr/local/bin/pkg-config (1.6.1)

backends/telepathy/lib/meson.build:19:0: ERROR: Static libraries can only be introspected with GObject-Introspection >=1.58.1
```

In meson-logs/meson-log.txt, meson said it ran pkg-config with an empty `PKG_CONFIG_PATH`.
```
Pkg-config binary for MachineChoice.BUILD is not cached.
Pkg-config binary missing from cross or native file, or env var undefined.
Trying a default pkg-config fallback at pkg-config
Trying pkg-config binary pkg-config for machine MachineChoice.BUILD at ['/usr/local/bin/pkg-config']
Found pkg-config: /usr/local/bin/pkg-config (1.6.1)
Determining dependency 'gobject-introspection-1.0' with pkg-config executable '/usr/local/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/local/bin/pkg-config --modversion gobject-introspection-1.0` -> 0
1.56.1
PKG_CONFIG_PATH: 
Called `/usr/local/bin/pkg-config --cflags gobject-introspection-1.0` -> 0
-I/usr/local/include/gobject-introspection-1.0 -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -pthread -I/usr/local/lib/libffi-3.2.1/include
PKG_CONFIG_PATH: 
Called `/usr/local/bin/pkg-config gobject-introspection-1.0 --libs` -> 0
-L/usr/local/lib -lgirepository-1.0 -lgobject-2.0 -lglib-2.0 -lintl
PKG_CONFIG_PATH: 
Called `/usr/local/bin/pkg-config gobject-introspection-1.0 --libs` -> 0
-L/usr/local/lib -lgirepository-1.0 -lgobject-2.0 -lglib-2.0 -lintl
```

**Comment by worldofpeace** [comment]
related to #5502 perhaps

**Comment by jtojnar** [comment]
When building folks, I was getting the error in 0.51.1:

```
backends/telepathy/lib/meson.build:19:0: ERROR: Static libraries can only be introspected with GObject-Introspection >=1.58.1
```

But it appears to have been fixed in 0.51.2. Bisect leads to c95244727437b5641fdb8e91d800bd4a9d741fd2 as a fix.

**Comment by lantw44** [comment]
Thanks, I confirmed that it has been fixed in 0.51.2. I think we can close this issue.
